### DISCUSSION TITLE
Is it possible to call existing JUnit tests with JMeter DSL?

### DISCUSSION THREAD

**Discussion by peat78** [abstracta/jmeter-java-dsl#198] [Q&A]
Hi all,

I want to call existing JUnit tests with JMeter DSL like in JMeter GUI with JUnit Request sampler.

Do I need to implement such (I guess new) sampler or do I have to use a jmx test plan and use it in the DSL?

Any other idea?

Thanks!
P.

**Comment by rabelenda** [comment]
Hi, good question!

I have never used the JUnit request element from JMeter.

I would like to know more on how are you using existing JUnit tests in combination with JMeter, to give you a good advice on how to proceed.

Without additional details, you may use jsr223 sampler with java lambda code to easily invoke any java code, but at the expense of portability for execution in remote engines (like BlazeMeter). [Here](https://github.com/abstracta/jmeter-java-dsl/discussions/196#discussioncomment-6252232) we are discussing some options to be able to run lambdas in remote engines.

Another possibility is using [test element wrapper](https://abstracta.github.io/jmeter-java-dsl/guide/#custom-or-yet-not-supported-test-elements) to use existing JUnit Request feature, but I think it would not be as seamless as using just java code. You may optionally implement a JMeter DSL extension, and contribute it to the tool, but I would like to know before more context on motivation, and review if aren't there any simpler solutions for that scenario.

Regards

**Comment by peat78** [comment]
Hi,
thanks for your answer!

The reason, or actually the goal, is to "reuse" existing integration tests for performance testing.

Anyway, I'm thinking about refactoring our ITs and extract the main logic to separate classes and use them on both sides (ITs and performance tests). I guess, that I have to select the "test element wrapper" approach mentioned above to achieve this, right?

I would be interested to implement a sampler to execute JUnit 5 tests...

**Comment by peat78** [comment]
...hm, btw, the example on how to build own custom sampler based on [test element wrapper](https://abstracta.github.io/jmeter-java-dsl/guide/#custom-or-yet-not-supported-test-elements) does not match the actual implementation (i'm using the latest version 1.12) and the class hierarchy is different. 

is there an example implementation?

**Reply by rabelenda** [reply]
You are right, the documentation was not properly updated. It should say `extends BaseSampler` instead of `extends DslSampler`. Thank you for pointing out the issue.

**Comment by peat78** [comment]
...never mind, i've implemented a custom sampler for executing JUnit 5 test cases (but in a different way as described in [test element wrapper](https://abstracta.github.io/jmeter-java-dsl/guide/#custom-or-yet-not-supported-test-elements)) and it works for me

**Reply by rabelenda** [reply]
How did you solve it? I think using jsr223Sampler would probably be the simplest, but maybe you found out other solutions that are worth to take into consideration for the rest of the community :).

**Reply by peat78** [reply]
...nope, no jsr223Sampler - takes a bit time to explain, however, if i'll get time from my company i will contribute to the project ;-)
